### ISSUE TITLE
[TS4.1 resolve's Parameters] not support

### ISSUE THREAD

**Issue by monic-shixi** [typescript-eslint/typescript-eslint#3219]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```TS
// fake.ts
const fake = () => {
  return new Promise(resolve => {
    resolve()
  })
}
```

with error:
```
Expected 1 arguments, but got 0. Did you forget to include 'void' in your type argument to 'Promise'?ts(2794)
lib.es2015.promise.d.ts(33, 34): An argument for 'value' was not provided.
```
Because of ts 4.1 breaking changes. And https://github.com/typescript-eslint/typescript-eslint/issues/2583 said no impact, so it will not be support?

**Expected Result**
error not show by eslint.

**Actual Result**
eslint show error and can be autofix.

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.18.0` |
| `@typescript-eslint/parser`        | `4.18.0` |
| `TypeScript`                       | `4.2.3` |
| `ESLint`                           | `7.14.0` |
| `node`                             | `12.13.1` |

**Comment by bradzacher** [comment]
This is a TS error, not an eslint error.

Note the error code of the message.

Eslint error codes will say `eslint(.....)`, but yours says `ts(.....)`

**Comment by monic-shixi** [comment]
> This is a TS error, not an eslint error.
> 
> Note the error code of the message.
> 
> Eslint error codes will say `eslint(.....)`, but yours says `ts(.....)`

Yep, i known. So, eslint will not support this syntax check?
